File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/IAccountSettingsDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO
{
    using WorldFeed.Common.Public.Models.Enums;

    // I am the stupidest person in the world
    public interface IAccountSettingsDTO
    {
        string ScreenName { get; set; }

        PrivacyMode PrivacyMode { get; set; }

        Language Language { get; set; }

        bool AlwaysUseHttps { get; set; }

        bool DiscoverableByEmail { get; set; }

        bool DiscoverableByMobilePhone { get; set; }

        bool DisplaySensitiveMedia { get; set; }

        bool SmartMute { get; set; }

        bool GeoEnabled { get; set; }

        bool UseCookiePersonalization { get; set; }

        AllowDirectMessagesFrom AllowDirectMessagesFrom { get; set; }

        AllowDirectMessagesFrom AllowGroupDirectMessagesFrom { get; set; }

        AllowContributorRequestMode AllowContributorRequest { get; set; }

        ITimeZone TimeZone { get; set; }

        bool SleepTimeEnabled { get; set; }

        int SleepTimeStartHour { get; set; }

        int SleepTimeEndHour { get; set; }
        
        string TranslatorType { get; set; }

        ITrendLocation[] TrendLocations { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/DTO/IAccountSettingsDTO.ts
import {PrivacyMode} from '../../Enum/PrivacyMode';
import {ITimeZone} from "../ITimeZone";
import {ITrendLocation} from "../ITrendLocation";
import { AllowDirectMessagesFrom } from '../../Enum/AllowDirectMessagesFrom';
import {AllowContributorRequestMode} from "../../Enum/AllowContributorRequestMode";
import { Language } from '../../Enum/Language';

export interface IAccountSettingsDTO {
  screenName: string;
  privacyMode: PrivacyMode;
  language: Language;

  alwaysUseHttps: boolean;
  discoverableByEmail: boolean;
  discoverableByMobilePhone: boolean;
  displaySensitiveMedia: boolean;
  smartMute: boolean;

  geoEnabled: boolean;
  useCookiePersonalization: boolean;

  allowDirectMessagesFrom: AllowDirectMessagesFrom;
  allowGroupDirectMessagesFrom: AllowDirectMessagesFrom;
  allowContributorRequest: AllowContributorRequestMode;

  timeZone: ITimeZone;

  sleepTimeEnabled: boolean;
  sleepTimeStartHour: number;
  sleepTimeEndHour: number;

  translatorType: string;
  trendLocations: ITrendLocation[];
}
